File: app/src/main/java/com/hijamoya/example/dagger/injection/ConductorInjectionModule.java
package com.hijamoya.example.dagger.injection;

import com.bluelinelabs.conductor.Controller;

import java.util.Map;

import dagger.Module;
import dagger.android.AndroidInjector;
import dagger.multibindings.Multibinds;

@Module
public abstract class ConductorInjectionModule {

    private ConductorInjectionModule() {
    }

    @Multibinds
    abstract Map<Class<? extends Controller>,
            AndroidInjector.Factory<? extends Controller>> controllerInjectorFactories();

    @Multibinds
    abstract Map<String, AndroidInjector.Factory<? extends Controller>>
    controllerInjectorFactoriesWithStringKeys();

}

File: app/src/main/kotlin/com/hijamoya/example/dagger/injection/ConductorInjectionModule.kt
package com.hijamoya.example.dagger.injection

import com.bluelinelabs.conductor.Controller

import dagger.Module
import dagger.android.AndroidInjector
import dagger.multibindings.Multibinds

@Module
abstract class ConductorInjectionModule private constructor() {

    @Multibinds abstract fun controllerInjectorFactories(): Map<Class<out Controller>,
            AndroidInjector.Factory<out Controller>>

    @Multibinds abstract fun controllerInjectorFactoriesWithStringKeys(): Map<String,
            AndroidInjector.Factory<out Controller>>
}
